tideways/provision.sh

```diff
@@ -0,0 +1,119 @@
+
+install_tideways() {
+    # Tideways is only for php =>7.0
+    if [[ ! -d /var/local/tideways-php/.git ]]; then
+        echo "Cloning Tideways extension"
+        git clone "https://github.com/tideways/php-xhprof-extension" /var/local/tideways-php
+    else
+        echo "Updating Tideways extension"
+        ( cd /var/local/tideways-php/ && git pull -q && git checkout -q )
+    fi
+}
+
+install_tideways_php() {
+    version=$1
+    if [[ `command -v php$version` ]]; then
+        echo "Copying tideways files for php $version"
+        cp -f "${DIR}/tideways.ini" "/etc/php/$version/mods-available/tideways_xhprof.ini"
+        cp -f "${DIR}/mongodb.ini" "/etc/php/$version/mods-available/mongodb.ini"
+        cp -f "${DIR}/xhgui-php.ini" "/etc/php/$version/mods-available/xhgui.ini"
+        cp -rf /var/local/tideways-php /var/local/tideways-php$version
+        echo "Compiling Tideways for PHP $version"
+        cd "/var/local/tideways-php${version}"
+        update-alternatives --set php /usr/bin/php$version > /dev/null 2>&1
+        update-alternatives --set php-config /usr/bin/php-config$version > /dev/null 2>&1
+        update-alternatives --set phpize /usr/bin/phpize$version > /dev/null 2>&1
+        phpize$version
+        ./configure --enable-tideways-xhprof --with-php-config=php-config$version > /dev/null 2>&1
+        make > /dev/null 2>&1
+        make install > /dev/null 2>&1
+    fi
+}
+
+restart_php() {
+    for version in 7.0 7.1 7.2 7.3
+    do
+        if [[ `command -v php$version` ]]; then
+            service php$version-fpm restart
+        fi
+    done
+    service nginx restart
+}
+
+install_mongodb() {
+    apt-key add "${DIR}/aptkey.pgp"
+    echo "deb http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-3.4.list
+    sudo apt update > /dev/null 2>&1
+    apt-get -y install mongodb-org re2c
+    sudo pecl install mongodb
+    ln -s /usr/lib/php/20151012/mongodb.so /usr/lib/php/20170718/mongodb.so
+    ln -s /usr/lib/php/20160303/mongodb.so /usr/lib/php/20170718/mongodb.so
+    ln -s /usr/lib/php/20180731/mongodb.so /usr/lib/php/20170718/mongodb.so
+    phpenmod mongodb
+    # auto-remove records older than 2592000 seconds (30 days)
+    mongo xhprof --eval 'db.collection.ensureIndex( { "meta.request_ts" : 1 }, { expireAfterSeconds : 2592000 } )' > /dev/null 2>&1
+    # indexes
+    mongo xhprof --eval  "db.collection.ensureIndex( { 'meta.SERVER.REQUEST_TIME' : -1 } )" > /dev/null 2>&1
+    mongo xhprof --eval  "db.collection.ensureIndex( { 'profile.main().wt' : -1 } )" > /dev/null 2>&1
+    mongo xhprof --eval  "db.collection.ensureIndex( { 'profile.main().mu' : -1 } )" > /dev/null 2>&1
+    mongo xhprof --eval  "db.collection.ensureIndex( { 'profile.main().cpu' : -1 } )" > /dev/null 2>&1
+    mongo xhprof --eval  "db.collection.ensureIndex( { 'meta.url' : 1 } )" > /dev/null 2>&1
+    update-rc.d mongodb defaults
+    update-rc.d mongodb enable
```

Comment:
I am getting also `update-rc.d: /etc/init.d/mongodb: file does not exist`

Reply:
+1, mongodb and tideways installation should be functionally separate